src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -269,163 +298,192 @@ default MemorySegment allocate(ValueLayout.OfDouble layout, double value) {
      * @param layout the layout of the block of memory to be allocated.
      * @param value  the value to be set in the newly allocated memory segment.
      */
-    default MemorySegment allocate(AddressLayout layout, MemorySegment value) {
+    default MemorySegment allocateFrom(AddressLayout layout, MemorySegment value) {
         Objects.requireNonNull(value);
         Objects.requireNonNull(layout);
-        MemorySegment seg = allocate(layout);
-        layout.varHandle().set(seg, value);
-        return seg;
+        MemorySegment segment = allocateNoInit(layout);
+        segment.set(layout, 0, value);
+        return segment;
+    }
+
+    /**
+     * {@return a new memory segment with a {@linkplain MemorySegment#byteSize() byteSize()} of
```

Comment:
```suggestion
     * {@return a new memory segment with a {@linkplain MemorySegment#byteSize() byte size} of
```

Reply:
Also, in the panama repo I see this:
```
Allocates a memory segment with the given layout and initializes it with the bytes in the provided source memory segment.
```
Which seems more correct - e.g. more consistent with other allocation methods, and also more succinct (note that the first sentence is really what shows up in the method summary javadoc, so there is a certain interest in providing a quick description of what the method does,

Reply:
Panama repo change:
https://github.com/openjdk/panama-foreign/commit/06e2017883c939188103c4dd53185417a00b2921

But, this code was altered in a follow up merge - maybe the merge was problematic?

Reply:
This was changed in the main line repo as a result of: https://github.com/openjdk/jdk/pull/14997 Since all the other methods were using `{@return ...}` I changed this new overload to that style as well.

Reply:
I don't mind changing it back to the old style, but I think the style should be consistent for all the allocateFrom overloads? So, I'd have to change all of them back.

Reply:
I forgot about the change that went into mainline. Do you have a link of the latest javadoc? I'd like to check how the method summary looks.

Reply:
Here you go: https://cr.openjdk.org/~jvernee/FFM_22_PR_v1/java.base/java/lang/foreign/SegmentAllocator.html#allocateFrom(java.lang.foreign.ValueLayout,java.lang.foreign.MemorySegment,java.lang.foreign.ValueLayout,long,long)

Reply:
Ok, now I'm more convinced that the method summary really does look bad (or worse, compared to 20).

For instance [allocateFrom](https://cr.openjdk.org/~jvernee/FFM_22_PR_v1/java.base/java/lang/foreign/SegmentAllocator.html#allocateFrom(java.lang.foreign.ValueLayout.OfByte,byte...):

```
Returns a new memory segment with a byteSize() initialized with the provided E byte elements as specified by the provided layout (i.e. byte ordering, alignment and size).
```

(same is true for all the other array-accepting `allocateFrom` methods). This should be simplified to:

```
Returns a new memory segment initialized with the elements in the provided byte array.
```
(then, if we want to say that the initialization honors the endianness of the provided layout, we can do so in a followup para, but the method summary should be simple).

So, once all the array-accepting methods are fixed, the segment-accepting `allocateFrom` needs to be simplified to:

```
Returns a new memory segment initialized with the contents of the provided segment.
```

Reply:
I've updated the 'header' of these methods, and instead added a more elaborated comment in a second paragraph. See: https://github.com/openjdk/jdk/pull/15103/commits/49bdd953eaed8732b59a803516355abc8df31fa3

Here is the re-generated javadoc: https://cr.openjdk.org/~jvernee/FFM_22_PR/v2/java.base/java/lang/foreign/SegmentAllocator.html#allocateFrom(java.lang.foreign.ValueLayout.OfInt,int...)